crates/valence_network/src/connect.rs

```diff
@@ -79,7 +79,15 @@ async fn handle_connection(
         error!("failed to set TCP_NODELAY: {e}");
     }
 
-    match try_handle_legacy_ping(&shared, &mut stream, remote_addr).await {
+    let timeout = Duration::from_secs(5);
+
+    match tokio::time::timeout(
```

Comment:
This made me realize there should be a single timeout where `handle_connection` is called instead of individual timeouts for every packet send/recv via `PacketIo`. that'll have to be fixed later.